group_vars/galaxy-test.yml

```diff
@@ -256,7 +256,7 @@ galaxy_user:
     name: galaxy
     create_home: true
     home: /opt/galaxy
-    uid: 999
+    uid: 991
```

Comment:
I'm not sure this will work. We need to have this, because the UID is shared between the other users on all VMs and Nodes if I remember correctly. @hexylena @sj213 maybe you can add more information here?

Reply:
yeah that's going to screw up permissions, that uid is used everywhere. Any reason to change this?

Reply:
Well, it's `-test`, so, should be better. but still there's stuff that might be affected, data directories, etc.

Reply:
polkitd is using that id by default. I can rearrange it.

Reply:
oh, yeah, that was a known issue. Just have to rename polkit, there's a role to do that even for the apollo VM (but it's kind of gross / ugly)

https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/apollo.yml#L38

Reply:
Uid/gid 999/999 for Galaxy was chosen back in the days of SL/CentOS 6.x, where system accounts' IDs where growing from 1 up and the end of the system range was never reached in practice. Along come systemd and OD and stuff and now some service IDs are growing _downwards_ from 999... And no, the Galaxy IDs can of course _not_ be changed just like that, not when using NFSv3 anyway. _In theory_, using NFSv4 _only_ (which uses user- and group-_names_ rather than IDs on the wire) would allow for different numerical IDs on client and server, but I **strongly** advise against this - It's error prone and Linux NFSv4-code has never been too reliable, and then there are players like the Isilon which may present additional complications wrt NFSv4.

As has been noted already, the uid/gid pair of a service running as 999/999 by default can be changed (provided the machine is booted or at least the service restarted after renumbering its files on disk - but when building a VM image this is implied anyway). For the CentOS-7 nodes of the Backofen-cluster I've tried to come up with a solution that is as safe and robust as possible, but it's still a hack of course:

https://github.com/BackofenLab/compute-image-playbook/tree/backofen/galaxy-user-std